go/teams/service_helper.go

```diff
@@ -120,6 +121,98 @@ func SetRoleReader(ctx context.Context, g *libkb.GlobalContext, teamname, userna
 	return ChangeRoles(ctx, g, teamname, keybase1.TeamChangeReq{Readers: []keybase1.UserVersion{uv}})
 }
 
+func tryToCompleteInvites(ctx context.Context, g *libkb.GlobalContext, team *Team, username string, uv keybase1.UserVersion, req *keybase1.TeamChangeReq) error {
+	if team.NumActiveInvites() == 0 {
+		return nil
+	}
+
+	getUserProofs := func() *libkb.ProofSet {
+		arg := keybase1.Identify2Arg{
+			UserAssertion:    username,
+			UseDelegateUI:    false,
+			Reason:           keybase1.IdentifyReason{Reason: "clear invitation when adding team member"},
+			CanSuppressUI:    true,
+			IdentifyBehavior: keybase1.TLFIdentifyBehavior_CHAT_GUI,
+			NeedProofSet:     true,
+		}
+		eng := engine.NewResolveThenIdentify2(g, &arg)
+		ectx := &engine.Context{
+			NetContext: ctx,
+		}
+		if err := engine.RunEngine(eng, ectx); err == nil {
+			return eng.GetProofSet()
+		}
+		return nil
+	}
+
+	proofs := getUserProofs()
+	if proofs == nil {
+		return fmt.Errorf("Cannot get proof set for user.")
+	}
+
+	var completedInvites = map[keybase1.TeamInviteID]keybase1.UserVersionPercentForm{}
+
+	for i, invite := range team.chain().inner.ActiveInvites {
+		g.Log.CDebugf(ctx, "tryToCompleteInvites invite %q %+v", i, invite)
+		ityp, err := invite.Type.String()
+		if err != nil {
+			return err
+		}
+		category, err := invite.Type.C()
+		if err != nil {
+			return err
+		}
+
+		if category != keybase1.TeamInviteCategory_SBS {
+			continue
+		}
+
+		proofsWithType := proofs.Get([]string{ityp})
+
+		var proof *libkb.Proof
+		for _, p := range proofsWithType {
+			if p.Value == string(invite.Name) {
+				proof = &p
+				break
+			}
+		}
+
+		if proof == nil {
+			continue
+		}
+
+		g.Log.CDebugf(ctx, "Found proof in user's ProofSet: key: %s value: %q; invite proof is %s@%s", proof.Key, proof.Value, string(invite.Name), ityp)
+
+		resolveResult := g.Resolver.ResolveFullExpressionNeedUsername(ctx, fmt.Sprintf("%s@%s", string(invite.Name), ityp))
+		g.Log.CDebugf(ctx, "Resolve result is: %+v", resolveResult)
+		if resolveResult.GetError() != nil || resolveResult.GetUID() != uv.Uid {
+			// Cannot resolve invitation or it does not match user
+			continue
+		}
+
+		assertion := fmt.Sprintf("%s@%s+uid:%s", string(invite.Name), ityp, uv.Uid)
+		g.Log.CDebugf(ctx, "Assertion is: %s", assertion)
+
+		arg := keybase1.Identify2Arg{
```

Comment:
Alright how about rebasing and using `verifyResolveResult` to validate the resolution. Would that fit?

Reply:
Tried to use new API, but I have a feeling I missed few things and it's still needlessly complicated. I'd appreciate if you can take a look. Thanks!